services/outputhost/consconnection.go

```diff
@@ -408,6 +399,15 @@ func (conn *consConnection) createMsgAndWriteToClientUtil(msg *cherami.ConsumerM
 			}
 		}
 	}
+	// Always add to the messageCache regardless of whether the write
+	// to the client succeeded or not. If we don't add it to the
+	// message cache, the message is *lost* and will never be redelivered.
+	// This can cause the CG to be stuck forever waiting for an ACK from client.
+	// The blocking operation below can delay the closing of stream when a
+	// conn close signal is received, the assumption here is that the
+	// operation if it blocks, will only do so for less than a second
+	msgCacheCh <- cacheMsg{msg: msg, connID: conn.connID}
```

Comment:
This is a valid assumption but we should add an alert on message cache layer if any of the "cases" there takes more than a second. We can do it as a followup patch. @GuillaumeBailey already measures the time it takes on each "case" there.